src/Appwrite/Auth/OAuth2/Disqus.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace Appwrite\Auth\OAuth2;
+
+use Appwrite\Auth\OAuth2;
+
+// Reference Material
+// https://disqus.com/api/docs/auth/
+
+class Disqus extends OAuth2
+{
+    private string $endpoint = 'https://disqus.com/api/';
+    protected array $user = [];
+    protected array $tokens = [];
+    protected array $scopes = [
+        "read",
+        "email",
+    ];
+
+    public function getName(): string
+    {
+        return "disqus";
+    }
+
+    public function getLoginURL(): string
+    {
+        $url = $this->endpoint . 'oauth/2.0/authorize/?' .
+            \http_build_query([
+                'response_type' => 'code',
+                'client_id' => $this->appID,
+                'state' => \json_encode($this->state),
+                'redirect_uri' => $this->callback,
+                'scope' => \implode(',', $this->getScopes())
+            ]);
+
+        return $url;
+    }
+
+    protected function getTokens(string $code): array
+    {
+        if (empty($this->tokens)) {
+            $this->tokens = \json_decode($this->request(
+                'POST',
+                $this->endpoint . 'oauth/2.0/access_token/',
+                ["Content-Type: application/x-www-form-urlencoded"],
+                \http_build_query([
+                    'grant_type' => 'authorization_code',
+                    "client_id" => $this->appID,
+                    "client_secret" => $this->appSecret,
+                    "redirect_uri" => $this->callback,
+                    'code' => $code,
+                    'scope' => \implode(' ', $this->getScopes()),
+                ])
+            ), true);
+        }
+        return $this->tokens;
+    }
+
+    public function refreshTokens(string $refreshToken): array
+    {
+        $this->tokens = \json_decode($this->request(
+            'POST',
+            $this->endpoint . 'oauth/2.0/access_token/?',
+            ['Content-Type: application/x-www-form-urlencoded'],
+            \http_build_query([
+                'grant_type' => 'refresh_token',
+                'refresh_token' => $refreshToken,
+                'client_id' => $this->appID,
+                'client_secret' => $this->appSecret,
+            ])
+        ), true);
+
+        if (empty($this->tokens['refresh_token'])) {
+            $this->tokens['refresh_token'] = $refreshToken;
+        }
+
+
+        return $this->tokens;
+    }
+
+    public function getUserID(string $accessToken): string
+    {
+        $user = $this->getUser($accessToken);
+
+        $userId = $user["response"]["id"];
+
+        return $userId;
+    }
+
+    public function getUserEmail(string $accessToken): string
+    {
+        $user = $this->getUser($accessToken);
+
+        $userEmail = $user["response"]["email"];
+
+        return $userEmail;
+    }
+
+    public function isEmailVerified(string $accessToken): bool
+    {
+        $user = $this->getUser($accessToken);
+
+        $isVerified = $user["response"]["isAnonymous"];
```

Comment:
Officially, disqus Oauth does not give a parameter which says that their email is verified. I have currently pointed it to `isAnonymous` for now. But I guess this is not correct.  Should we omit this?

Reply:
unless OAuth clearly states the parameter describes email verification status, we should return `false`,

Reply:
Hmmm but I just registered and got a verification email 🤔 Can we please double-check all docs to make sure there isn't a parameter for it?

Reply:
Disqus gives a verification email but there isn't any parameter for it. I have double-checked the docs. Maybe the endpoint is still in Beta and they may provide a `verification` parameter in the future?

I will change it to false for now.